hibernate-core/src/test/java/org/hibernate/jpa/test/exception/ExceptionTest.java

```diff
@@ -111,8 +111,7 @@ public void testConstraintViolationException() throws Exception {
 			fail();
 		}
 		catch ( PersistenceException e ) {
-			Throwable t = e.getCause();
-			assertTrue( "Should be a constraint violation", t instanceof ConstraintViolationException );
+			assertTrue( "Should be a constraint violation", e instanceof ConstraintViolationException );
```

Comment:
this breaks 5.1 compatibility

Reply:
Does it?

It breaks 5.2 and early 5.3 compatibility for sure but the idea is to restore the 5.1 behavior.